### ISSUE TITLE
Child tabs in tree are not indented

### ISSUE THREAD

**Issue by nanoscopic** [piroor/treestyletab#2985]
## Short description
Tabs in tree are not indented

## Steps to reproduce

 1. Start Firefox with clean profile.
 2. Install TST.
 3. Open multiple tabs, one after another, following links from within each to create a tree

## Expected result
Tabs are indented.

## Actual result
Tabs are not indented.

## Environment
This occurs of 3 different machines I have. Two of them run Windows. One is MacOS.
It essentially occurs on all updated Firefox + updated TST.
Firefox 90.0.2 on Mac has the issue.
Firefox 91.0 on Windows has the issue.

It appears to happen in all updated versions of TST. ( 3.8.9 )

 * Platform (OS): Windows and MacOS
 * Version of Firefox: 90.0.2, 91.0
 * Version (or revision) of Tree Style Tab: 3.8.9

**Comment by nanoscopic** [comment]
This bug appears to be caused by this change: https://github.com/piroor/treestyletab/commit/b6231a9d07db8112e9a1a836b9787ad1cf7d7ee4

**Comment by piroor** [comment]
I cannot reproduce the problem with a clean profile, TST 3.8.9 on Nightly 93.0a on Windows 10. Could you narrow down conditions to reproduce?

**Comment by nanoscopic** [comment]
I can reproduce yes.

Brand new install on another MacOS machine. Firefox 91. TST 3.8.9.

Steps:
1. Install brand new browser ( normal released version 91 ) ; guessing version is irrelevant though.
2. Install TST ( installs 3.8.9 )
3. Open a few tabs in a tree ( they look normal and are indented )
4. Moved tree to the right side as I prefer.
5. Set browser to restore previous session
6. Exit browser
7. Reopen browser
8. Tabs are no longer indented.

**Comment by Bert-Proesmans** [comment]
The same thing is happening here without creating a new profile. I noticed today that the child tab alignment is different.  
I haven't changed any TST config since a few weeks ago where my custom CSS code was borked due to dom tree changes within the extension code (there was a GitHub issue but can't remember what to look for exactly)

Below is a screenshot of an expanded tree root with one child. The arrow indicates expansion but the child item stays on the same level. I restarted the browser to verify it wasn't a temporary bug, and it's consistently reproducible.
![afbeelding](https://user-images.githubusercontent.com/2027877/129788013-d1e1ec91-347c-442b-bd7b-8ad2b0bf3efc.png)

<details>
<summary>Specs</summary>
TST V3.8.9 (2021-08-16)  

FF 91.0.1  

Win10  

</details>

**Comment by piroor** [comment]
Ah, OK I got it. It is expected behavior and tabs are actually indented **with right margins** due to the effect of the option "Style of contents for the sidebar position: Right side / Auto detect" but hard to see with the Proton theme if tabs have short titles.

Screenshot with "Proton":
![image](https://user-images.githubusercontent.com/70062/129830244-a0c126de-02e7-4d2c-a87c-8df3e3d7fdc6.png)
Screenshot with "Photon":
![image](https://user-images.githubusercontent.com/70062/129830319-a1f33674-f573-43b0-b01b-63912ef82032.png)

Thus, there are some possible workarounds:

* Choose TST Options => Appearance => Style of contents for the sidebar position => "Left side". Then tabs become indented with left margins always.
* Installing a helper addon "TST Indent Line" https://addons.mozilla.org/firefox/addon/tst-indent-line/ for better visibility of indentations. Screenshot:
  ![image](https://user-images.githubusercontent.com/70062/129830565-18873e64-ffa1-431d-97df-4c4fc9655fcd.png)

**Comment by nanoscopic** [comment]
As I documented above, the tabs are indented correctly on startup and only look different after a restart. There is more to it than that.

I do agree that changing the appearance options does "correct it" to appear as it did previously, but it is a huge change to default to the new look compared to how it looked previously with the default settings.

The default settings should lead it to looking the way it did previously. ( indented from left side in a visible fashion )

Also, your "screenshot with photon" shows a darker background in the indent space. This does not occur for me unless I set the style to "high contrast".

Thank you though for pointing out how to make it look the way it did previously. I did look at the options and it was not obvious at all what was going on. The graphics shown in the options aren't representative of the results.

**Comment by piroor** [comment]
> As I documented above, the tabs are indented correctly on startup and only look different after a restart. There is more to it than that.

I think it happened because you didn't reopen the sidebar panel after you moved the sidebar to right. TST won't update its "left side or right side" appearance until the sidebar panel is reopened because there is no handlable event fired when the sidebar position is changed. So TST detected the "next startup after restarting" as "the initial time the sidebar panel is loaded in the rightside positioned sidebar" on your case.

Sorry I forgot to introduce any migration process about the change, thus I lost the last chance to keep the default choice on the previous version for users of old versions. As a workaround I implemented a confirmation on the initial time TST detects the sidebar is opened at the right at #2984. I hope it helps to reduce the confusion.

----

Basically I believed that most people who put the sidebar at right use the "Right side" appearance of TST. I couldn't imagine how much people used the "Left side" appearance for the sidebar placed at right...

**Comment by piroor** [comment]
Just an apology: the current default behavior "TST's sidebar panel is shown with its leftside appearance in the left positioned sidebar, rightside appearance in the right positioned sidebar" was a behavior I begrudgingly dropped at the migration from TST 0.x to TST 2.0 (4 years ago). I gave up to reintroduce the auto detection for a long time because I couldn't find out how to do that, and I finally did that at 3.8.9 with a technique I recently knew. Thus I thought "TST 2.0-3.8.8 were broken and TST 3.8.9 gets back the old behavior".

However everything changed while in this 4 years. The default theme was changed to Proton (no border so it has bad user experience with TST's rightside appearance), and people who used TST 0.x left from Firefox (they look to use Palemoon or other browsers based on old Firefox). Today, the number of people who use TST with "rightside positioned sidebar + leftside appearance" looks larger than I thought. Now I totally regret the change "auto detection of the appearance for the current sidebar position"...

**Comment by nanoscopic** [comment]
Thanks for all your detailed responses. The reason I point out it should remain the same as for the last several years when the sidebar is positioned on the right is only because having tree styles tabs is one of the main reasons I ( and many others ) use Firefox instead of Chrome.

Many many people, including myself, really greatly appreciate this add on. Thank you very much for making it.

The only reason I ask it remain the same is because I am sure keeping it as before will reduce how many people bug you about "why doesn't it work anymore".

Best of luck and I hope you are able to easily tweak it so that people aren't confused and bothering you.

Thanks again for all your work.

**Comment by Bert-Proesmans** [comment]
Works as expected, for me, with the newest update installed!
Thanks for the elaborate explanation and quick solution 👍
